src/gasPriceOracle/adapters/solana.ts

```diff
@@ -0,0 +1,34 @@
+import { Provider } from "../../arch/svm";
+import { toBN, dedupArray } from "../../utils";
+import { GasPriceEstimate } from "../types";
+import { GasPriceEstimateOptions } from "../oracle";
+import { CompilableTransactionMessage, TransactionMessageBytesBase64, compileTransaction } from "@solana/kit";
+
+/**
+ * @notice Returns result of getFeeForMessage and getRecentPrioritizationFees RPC calls.
+ * @returns GasPriceEstimate
+ */
+export async function messageFee(provider: Provider, opts: GasPriceEstimateOptions): Promise<GasPriceEstimate> {
+  const { unsignedTx: _unsignedTx } = opts;
+
+  // Cast the opaque unsignedTx type to a solana-kit CompilableTransactionMessage.
+  const unsignedTx = _unsignedTx as CompilableTransactionMessage;
```

Comment:
It'd be nice to verify this via superstruct - then we'd inherit type inference as well.